### PR TITLE
[Feature] Support set password in new RBAC privilege framework

### LINKED ISSUE #15045: Support set password, set user property, alter user set password in new RBAC privilege framework
## Feature request

**Is your feature request related to a problem? Please describe.**

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by waittttting**
## What type of PR is this：
- [ ] BugFix
- [x] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #15045 

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [ ] I have checked the version labels which the pr will be auto backported to target branch
  - [ ] 2.5
  - [ ] 2.4
  - [ ] 2.3
  - [ ] 2.2

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 13 / 14 (92.86%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/qe/SetExecutor.java|10|11|90.91%|[89]|
|:large_blue_circle:|com/starrocks/authentication/AuthenticationManager.java|3|3|100.00%|[]|

**Review by gengjun-git — APPROVED** [review]

File: fe/fe-core/src/main/java/com/starrocks/authentication/AuthenticationManager.java [hunk 1]
```diff
@@ -224,6 +224,11 @@ public void createUser(CreateUserStmt stmt) throws DdlException {
     public void alterUser(AlterUserStmt stmt) throws DdlException {
         UserIdentity userIdentity = stmt.getUserIdent();
         UserAuthenticationInfo info = stmt.getAuthenticationInfo();
+        upgradeUserWithAuthenticationInfoUnlocked(userIdentity, info);
+    }
+
+    public void upgradeUserWithAuthenticationInfoUnlocked(UserIdentity userIdentity,
```
Comment (nshangyiming) [inline fe/fe-core/src/main/java/com/starrocks/authentication/AuthenticationManager.java:230]: upgrade -> update
remove Unlocked, this is actually locked

**Review by nshangyiming — COMMENTED** [review]
left one comment

**Review by Astralidea — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Pull request merged**